client/components/OperationList.js

```diff
@@ -71,6 +71,65 @@ class CategorySelectComponent extends React.Component {
     }
 }
 
+class OperationTypeSelectComponent extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            editMode: false
+        };
+    }
+
+    dom() {
+        return this.refs.cat.getDOMNode();
+    }
+
+    onChange(e) {
+        var selectedId = this.dom().value;
+        Actions.SetOperationType(this.props.operation, selectedId);
+        // Be optimistic
+        this.props.operation.type = selectedId;
+    }
+
+    switchToEditMode() {
+        this.setState({ editMode: true }, function() {
+            this.dom().focus();
+        });
+    }
+
+    switchToStaticMode() {
+        this.setState({ editMode: false });
+    }
+
+    render() {
+        var selectedId = this.props.operation.type;
+        var label = store.operationTypeToLabel(selectedId);
+
+        if (!this.state.editMode) {
+            return (<span onClick={this.switchToEditMode.bind(this)}>{label}</span>)
+        }
+
+        // On the first click in edit mode, categories are already loaded.
+        // Every time we reload categories, we can't be in edit mode, so we can
+        // just synchronously retrieve categories and not need to subscribe to
+        // them.
+        var options = store.getOperationTypes().map(function(c) {
+            if (c.id == NONE_OPERATION_TYPE_ID) {
```

Comment:
Is the intent to block the user from setting "None" as the operation type? In that case, you might as well go with:

```
var options = store.getOperationTypes().filter((t) => t.id !== NONE_OPERATION_TYPE_ID).map(...)
```

and the map doesn't need to contain the special case with the `if`.